src/main/java/com/example/demo/services/interfaces/IItemService.java

```diff
@@ -19,5 +19,5 @@
 	Collection<ItemModel> getActiveItemsByCategory(int categoryId, int page, int count);
 	Collection<ItemModel> getNewArrivalItems(int page, int count);
 	Collection<ItemModel> getLastChanceItems(int page, int count);
-	Collection<ItemModel> findItemsValidFilterCategories(String term, List<Integer> categories, int page, int count)throws InvalidDataException;
+	Collection<ItemModel> findItemsValidFilterCategories(String term, List<Integer> categories, int page, int count, String sort)throws InvalidDataException;
```

Comment:
Lets refactor these repeating params int page and int count with object

PaginationParams
lets then extend that with sorting
SortingPaginationParams